source/Nevermore/Advanced/Queryable/NevermoreQueryableExtensions.cs

```diff
@@ -32,6 +34,33 @@ public static IQueryable<TSource> WhereCustom<TSource>(this IQueryable<TSource>
                     source.Expression, Expression.Constant(whereClause)));
         }
 
+        public static IQueryable<TSource> Hint<TSource>(this IQueryable<TSource> source, string hint)
+        {
+            if (source == null)
+                throw new ArgumentNullException(nameof(source));
+            if (string.IsNullOrEmpty(hint))
+                throw new ArgumentNullException(nameof(hint));
+
+            return source.Provider.CreateQuery<TSource>(
+                Expression.Call(
+                    null,
+                    HintMethodInfo.MakeGenericMethod(typeof(TSource)),
+                    source.Expression, Expression.Constant(hint)));
+        }
+
+        public static string RawDebugView<TSource>(this IQueryable<TSource> source)
```

Comment:
The debug functionality can be simplified by taking it out of the translation/execution process and instead allowing this extension to call `Translate()` on the IQueryable's Provider, assuming it is the right type. Something like:

```
if (source.Provider is QueryProvider queryProvider)
{
    var (command, _) = queryProvider.Translate(source.Expression);
    return command.Statement;
}

throw new NotSupportedException();
```

Reply:
Much simpler, thanks for that. I've removed the debug code from the query builder down too.